test/integration/components/index.spec.ts

```diff
@@ -1,10 +1,21 @@
 import * as path from 'path';
+import { ToolpadEditor } from '../../models/ToolpadEditor';
 import { ToolpadHome } from '../../models/ToolpadHome';
 import { ToolpadRuntime } from '../../models/ToolpadRuntime';
-import { test } from '../../playwright/test';
+import { FrameLocator, Page, test } from '../../playwright/test';
 import { readJsonFile } from '../../utils/fs';
 
-test('components', async ({ page }) => {
+async function waitForComponents(page: Page | FrameLocator) {
+  await page.locator('text="foo button"').waitFor({ state: 'visible' });
+  await page.locator('img[alt="foo image"]').waitFor({ state: 'attached' });
+  await page.locator('text="foo datagrid column"').waitFor({ state: 'visible' });
+  await page.locator('text="custom component 1"').waitFor({ state: 'visible' });
+  await page.locator('label:has-text("foo textfield")').waitFor({ state: 'visible' });
+  await page.locator('text="foo typography"').waitFor({ state: 'visible' });
+  await page.locator('label:has-text("foo select")').waitFor({ state: 'visible' });
+}
+
+test('components', async ({ page, browserName }) => {
```

Comment:
Maybe the test name could describe its intention better, could be something like "components render correctly"?